File: wjsdncl/973_k_closest_points_to_origin.js
/**
 * 21m 16s 소요
 *
 * 시간복잡도: O(n log n)
 *
 * 1. 배영의 요소를 x^2 + y^2를 기준으로 정렬합니다.
 * 2. 가장 k개의 요소를 반환합니다.
 */
var kClosest = function (points, k) {
  return points.sort((a, b) => a[0] ** 2 + a[1] ** 2 - (b[0] ** 2 + b[1] ** 2)).slice(0, k);
};

File: citrussoda/7week/973_k-closest-points-to-origin.cpp
class Solution {
 public:
  vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
    sort(points.begin(), points.end(), [](auto& p1, auto& p2) {
      return p1[0] * p1[0] + p1[1] * p1[1] < p2[0] * p2[0] + p2[1] * p2[1];
    });

    points.resize(k);

    return points;
  }
};